src/plugProjectNishimuraU/GreenKabuto.cpp

```diff
@@ -1,399 +1,66 @@
-GreenKabuto::Obj::Obj()
-{
-	/*
-	stwu     r1, -0x10(r1)
-	mflr     r0
-	stw      r0, 0x14(r1)
-	extsh.   r0, r4
-	stw      r31, 0xc(r1)
-	mr       r31, r3
-	beq      lbl_802FFC04
-	addi     r0, r31, 0x2e4
-	lis      r3, __vt__Q24Game10PelletView@ha
-	stw      r0, 0x17c(r31)
-	addi     r3, r3, __vt__Q24Game10PelletView@l
-	li       r0, 0
-	stw      r3, 0x2e4(r31)
-	stw      r0, 0x2e8(r31)
-	stw      r0, 0x2ec(r31)
-
-lbl_802FFC04:
-	mr       r3, r31
-	li       r4, 0
-	bl       __ct__Q34Game6Kabuto3ObjFv
-	lis      r3, __vt__Q34Game11GreenKabuto3Obj@ha
-	addi     r0, r31, 0x2e4
-	addi     r5, r3, __vt__Q34Game11GreenKabuto3Obj@l
-	mr       r3, r31
-	stw      r5, 0(r31)
-	addi     r4, r5, 0x1b0
-	addi     r5, r5, 0x31c
-	stw      r4, 0x178(r31)
-	lwz      r4, 0x17c(r31)
-	stw      r5, 0(r4)
-	lwz      r4, 0x17c(r31)
-	subf     r0, r4, r0
-	stw      r0, 0xc(r4)
-	lwz      r31, 0xc(r1)
-	lwz      r0, 0x14(r1)
-	mtlr     r0
-	addi     r1, r1, 0x10
-	blr
-	*/
-}
+Obj::Obj() { }
 
 /*
  * --INFO--
  * Address:	802FFC58
  * Size:	0001A4
  */
-void GreenKabuto::Obj::changeMaterial()
+void Obj::changeMaterial()
 {
-	/*
-	stwu     r1, -0x20(r1)
-	mflr     r0
-	stw      r0, 0x24(r1)
-	stmw     r27, 0xc(r1)
-	mr       r27, r3
-	lwz      r3, 0x180(r3)
-	lwz      r4, 0x174(r27)
-	lwz      r12, 0(r3)
-	lwz      r29, 8(r4)
-	lwz      r12, 0xe0(r12)
-	lwz      r30, 4(r29)
-	mtctr    r12
-	bctrl
-	mr       r31, r3
-	mr       r3, r29
-	lwz      r12, 0(r29)
-	lwz      r12, 0x14(r12)
-	mtctr    r12
-	bctrl
-	lwz      r4, 0x174(r27)
-	lis      r3, j3dSys@ha
-	lbz      r0, 0(r31)
-	addi     r27, r3, j3dSys@l
-	lwz      r3, 8(r4)
-	li       r28, 0
-	lwz      r3, 4(r3)
-	lwz      r4, 0x6c(r3)
-	lwz      r3, 4(r4)
-	stb      r0, 0(r3)
-	lbz      r0, 1(r31)
-	stb      r0, 1(r3)
-	lhz      r0, 2(r31)
-	sth      r0, 2(r3)
-	lhz      r0, 4(r31)
-	sth      r0, 4(r3)
-	lbz      r0, 6(r31)
-	stb      r0, 6(r3)
-	lbz      r0, 7(r31)
-	stb      r0, 7(r3)
-	lbz      r0, 8(r31)
-	stb      r0, 8(r3)
-	lbz      r0, 9(r31)
-	stb      r0, 9(r3)
-	lhz      r0, 0xa(r31)
-	sth      r0, 0xa(r3)
-	lwz      r0, 0xc(r31)
-	stw      r0, 0xc(r3)
-	lbz      r0, 0x10(r31)
-	stb      r0, 0x10(r3)
-	lbz      r0, 0x11(r31)
-	stb      r0, 0x11(r3)
-	lbz      r0, 0x12(r31)
-	stb      r0, 0x12(r3)
-	lbz      r0, 0x13(r31)
-	stb      r0, 0x13(r3)
-	lbz      r0, 0x14(r31)
-	stb      r0, 0x14(r3)
-	lbz      r0, 0x15(r31)
-	stb      r0, 0x15(r3)
-	lbz      r0, 0x16(r31)
-	stb      r0, 0x16(r3)
-	lbz      r0, 0x17(r31)
-	stb      r0, 0x17(r3)
-	lbz      r0, 0x18(r31)
-	stb      r0, 0x18(r3)
-	lbz      r0, 0x19(r31)
-	stb      r0, 0x19(r3)
-	lha      r0, 0x1a(r31)
-	sth      r0, 0x1a(r3)
-	lwz      r0, 0x1c(r31)
-	stw      r0, 0x1c(r3)
-	lwz      r3, 4(r4)
-	lwz      r0, 0x1c(r3)
-	add      r0, r31, r0
-	subf     r0, r3, r0
-	stw      r0, 0x1c(r3)
-	lwz      r3, 4(r4)
-	lwz      r0, 0xc(r3)
-	add      r0, r31, r0
-	subf     r0, r3, r0
-	stw      r0, 0xc(r3)
-	b        lbl_802FFDD8
+	J3DModelData* modelData;
```

Comment:
Order of variables matters in this function? If not, then this can be moved to use

Reply:
Order changes codegen here, so they can't be moved down unfortunately (as ugly as it is)